### PR TITLE
108 create rl map cellphonedb input checks and readins

### PR DESCRIPTION
**Pull request by dimalvovs**
refactored:
- input checks;
- readins;
- python/R bool conversions;

### CODE REVIEW

**Review by jmitchell81 — APPROVED** [review]

File: R/import_fxns.R [hunk 1]
```diff
@@ -29,90 +29,36 @@ NULL
...
-  stopifnot(`Gene conversion host must be a string` = is(gene_conv_host, "character") & length(gene_conv_host) ==
-    1)
-  stopifnot(`Alternate conversion argument (not recommended) must be TRUE or FALSE` = is(
-    alternate_convert,
-    "logical"
-  ))
-  if (alternate_convert & is.null(alternate_convert_table)) {
-    stop("If using alternate conversion table (not recommended), a table must be provided")
-  }
+  check_arg(genes, c("character", "data.frame"))
```
Comment (drbergman) [inline R/import_fxns.R:34]: it looks like the `read_if_char` function acts on `genes` below in a way that requires `genes` to be a single character. So, should this also include a check as below of `allow_len = c(1)`?

File: R/import_fxns.R [hunk 2]
```diff
@@ -29,90 +29,36 @@ NULL
...
-    for (x in colnames(complexes)) {
-      if (identical(unique(complexes[[x]]), c("True", "False")) | identical(
-        unique(complexes[[x]]),
-        c("False", "True")
-      )) {
-        complexes[[x]] <- ifelse(complexes[[x]] == "True", TRUE, FALSE)
-      }
-    }
-  }
+  genes <-  conv_py_bools(genes)
```
Comment (drbergman) [inline R/import_fxns.R:57]: Any concern that any of these might contain all `True`'s or all `False`'s? It looks like `conv_py_bools` will only make changes if both are present.

**Review by drbergman — COMMENTED** [review]
Just two comments. Looks much simpler to read now!

File: R/import_fxns.R (hunk 1 above)
Comment (dimalvovs) [reply R/import_fxns.R:34]: valid point, addressed in [bd01e01](https://github.com/FertigLab/domino2/pull/109/commits/bd01e018f77d4aa58d6d4a017f7247923d228078)

File: R/import_fxns.R (hunk 2 above)
Comment (dimalvovs) [reply R/import_fxns.R:57]: @jmitchell81 could there be cases with only True or only False that we need to care about?

**Pull request merged**